core/internal/runmetadata/runmetadata.go

```diff
@@ -157,7 +157,32 @@ func InitRun(
 	defer operation.Finish()
 	ctx := operation.Context(metadata.beforeRunEndCtx)
 
-	// TODO: Update metadata if resuming, rewinding or forking.
+	// If resuming, rewinding or forking, we need to modify metadata
+	// in special ways before upserting the run.
+	switch {
+	case params.Settings.GetResume() != "":
+		err := metadata.updateMetadataForResume(params.Settings.GetResume())
+
+		if err != nil {
+			return nil, runUpdateErrorFromBranchError(err)
+		}
+
+	case params.Settings.GetResumeFrom() != nil:
+		err := metadata.updateMetadataForRewind(params.Settings.GetResumeFrom())
+
+		if err != nil {
+			return nil, runUpdateErrorFromBranchError(err)
+		}
+
+		// TODO: Originally we didn't upsert, but that seems like a bug.
```

Comment:
i think it is fine, based on the python implementation:        https://github.com/wandb/wandb/blob/main/wandb/sdk/internal/sender.py#L1019

Reply:
Thank you, removed the TODO!